spec/enumerable_spec.rb

```diff
@@ -0,0 +1,113 @@
+require_relative '../enumerables'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3] }
+  let(:hash) { { 'a' => 'enum', 'b' => 'module', 'c' => 'spec' } }
+
+  describe '#my_each' do
+    it 'return each value in an array' do
+      i = 0
+      array.my_each do |item|
+        expect(item).to eql(array[i])
+        i += 1
+      end
+    end
+
+    it 'return each value in a hash' do
+      hash.my_each do |key, value|
+        expect(value).to eql(hash[key])
+      end
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'returns a value from an index in an array' do
+      i = 0
+      array.my_each_with_index do |value, index|
+        expect(index).to eql(i)
+        expect(value).to eql(array[i])
+        i += 1
+      end
+    end
+  end
+
+  describe '#my_select' do
+    it 'return a new array based on block provided' do
+      expect(array.my_select(&:positive?)).to eql([1, 2, 3])
+    end
+
+    it 'return a new hash based on block provided' do
+      expect(hash.my_select { |_key, value| value == 'enum' }).to eql([%w[a enum]])
+    end
+  end
```

Comment:
In `#my_select `method. You have written for the correct usage and positive scenario of the method which is, `it 'returns an array based on block provided'`. Now write a test for a negative scenario, which is, `it 'does not return an array when no block is provided'`. Do the same for the hash check.